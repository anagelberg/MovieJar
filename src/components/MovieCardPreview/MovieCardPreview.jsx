import "./MovieCardPreview.scss";

function MovieCardPreview({ movie, setSelectedMovie, selected }) {
  return (
    <div className={selected ? "card card--selected" : "card"} onClick={() => { setSelectedMovie(movie) }}>
      <div className="card__overlay"></div>
      <img className="card__poster-img" src={`https://image.tmdb.org/t/p/original${movie.poster_path}`} alt="" />

    </div>
  );
}

export default MovieCardPreview;
