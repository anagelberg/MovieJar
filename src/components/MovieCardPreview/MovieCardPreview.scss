@use "../../styles/typefaces" as *;
@use "../../styles/variables" as *;
@use "../../styles/mixins" as *;

.card {
  width: 60px;
  height: 90px;
  position: relative;
  opacity: .9;

  @include tablet {
    width: 180px;
    height: 270px;
  }

  &__poster-img {
    object-fit: cover;
    width: 100%;
    height: 100%;
    position: center center;
    border-radius: $border-radius;
  }



  &:hover {
    border: 1px $primary solid;
    opacity: 1;
  }

  &--selected {
    border: 3px $accent solid;
    opacity: 1;
    &:hover {
      border: 3px $accent solid;
    }
  }
}
