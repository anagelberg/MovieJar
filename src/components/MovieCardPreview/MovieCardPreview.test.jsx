import { render, screen } from '@testing-library/react';
import MovieCardPreview from './MovieCardPreview';


const testMovie = {
    title: "The Godfather",
    poster_path: '/3bhkrj58Vtu7enYsRolD1fZdja1.jpg'
}


test('renders MovieCardPreview component', () => {
    render(<MovieCardPreview movie={testMovie} />);
});


test('MovieCardPreview component matches snapshot', () => {
    const { container } = render(<MovieCardPreview movie={testMovie} />);
    expect(container.firstChild).toMatchSnapshot();
});


test('displays correct movie information in MovieCardPreview component', () => {
    render(<MovieCardPreview movie={testMovie} />);
    const posterImage = screen.getByRole('img');
    expect(posterImage).toHaveAttribute('src', `https://image.tmdb.org/t/p/original${testMovie.poster_path}`);
});


test('MovieCardPreview click function called when clicked', () => {
    const setSelectedMovie = jest.fn();
    render(<MovieCardPreview movie={testMovie} setSelectedMovie={setSelectedMovie} />);
    screen.getByRole('img').click();
    expect(setSelectedMovie).toHaveBeenCalled();
});


test('MovieCardPreview component applies correct CSS class when selected', () => {
    let { container, rerender } = render(<MovieCardPreview movie={testMovie} selected={true} />);
    expect(container.firstChild).toHaveClass('cardPrev--selected');

    rerender(<MovieCardPreview movie={testMovie} selected={false} />);
    expect(container.firstChild).not.toHaveClass('cardPrev--selected');
})


test('renders MovieCardPreview null when no movie is provided', () => {
    const setSelectedMovie = jest.fn();
    const { container } = render(<MovieCardPreview setSelectedMovie={setSelectedMovie} />);
    expect(container.firstChild).toBeNull();
});



