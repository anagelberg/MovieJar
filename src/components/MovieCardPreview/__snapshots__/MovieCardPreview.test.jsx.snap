// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCardPreview component matches snapshot 1`] = `
<div
  class="cardPrev"
>
  <div
    class="card__overlay"
  />
  <img
    alt=""
    class="cardPrev__poster-img"
    src="https://image.tmdb.org/t/p/original/3bhkrj58Vtu7enYsRolD1fZdja1.jpg"
  />
</div>
`;
